src/Data/Nat/Primality.agda

```diff
@@ -141,3 +141,18 @@ private
   -- Example: 6 is composite
   6-is-composite : Composite 6
   6-is-composite = from-yes (composite? 6)
+
+------------------------------------------------------------------------
+-- Other properties
+
+Prime⇒NonZero : Prime n → NonZero n
+Prime⇒NonZero {suc _} _ = _
+
+Composite⇒NonZero : Composite n → NonZero n
+Composite⇒NonZero {suc _} _ = _
+
+-- If m is a factor of prime p, then it is equal to either 1 or p itself
+∣p⇒≡1∨≡p : ∀ m {p} → Prime p → m ∣ p → m ≡ 1 ⊎ m ≡ p
+∣p⇒≡1∨≡p 0 {p} p-prime m∣p = contradiction (0∣⇒≡0 m∣p) λ p≡0 → subst Prime p≡0 p-prime
+∣p⇒≡1∨≡p 1 {p} p-prime m∣p = inj₁ refl
+∣p⇒≡1∨≡p (suc (suc m)) {suc (suc p)} p-prime m∣p = inj₂ (≤∧≮⇒≡ (∣⇒≤ m∣p) λ m<p → p-prime (s≤s (s≤s z≤n)) m<p m∣p)
```

Comment:
Similarly there might be another useful lemma hiding here?

Reply:
`Prime⇒Irreducible`?

Reply:
That is, can we please have (somewhere) the definition
```agda
Irreducible p = ∀ {m} → m ∣ p → m ≡ 1 ⊎ m ≡ p
```
and then `∣p⇒≡1∨≡p` gets renamed to `prime⇒irreducible`, possibly even of *type* `Prime ⇒ Irreducible` for `Unary` predicate implication? (there's some explicit/implicit choreography here, but otherwise...)

Reply:
I _think_ (please correct me if I'm wrong) that that's the ring-theoretic definition of `Prime`, not `Irreducible`. Renaming `Prime` to `Irreducible` and introducing a corresponding `Prime` is an option but it's not without its costs.

Reply:
See #2180